### PR TITLE
Fix nightly pr failures

### PR DESCRIPTION
**Pull request by pb8o**
## Changes

PRs #4149 and #4188 had a few regressions bugs that were not reachable from the normal PR tests.

In addition, add a change so we only compile example binaries once, instead of once per worker. This avoids downloading and compiling code.

## Reason

To make all pipelines pass.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/integration_tests/security/test_vulnerabilities.py
Comment (sudanl0) [inline tests/integration_tests/security/test_vulnerabilities.py:183]: is this a temporary fix till we find why checker returns 2?

**Review by sudanl0 — APPROVED** [review]

**Review by ShadowCurse — APPROVED** [review]

File: tests/integration_tests/security/test_vulnerabilities.py
Comment (bchalios) [reply tests/integration_tests/security/test_vulnerabilities.py:183]: @zulinx86 do you have context around this?

**Review by bchalios — CHANGES_REQUESTED** [review]
I think it's important to answer @sudanl0's question regarding the workaround we use.

File: tests/integration_tests/security/test_vulnerabilities.py
Comment (pb8o) [reply tests/integration_tests/security/test_vulnerabilities.py:183]: I lack the context. In the A/B version of the test we ignore the return codes, so it would make sense to do it here too. I agree we need to figure out if ignoring is the right thing to do, but it could be done as a follow up, while we fix the pipelines.

Comment (zulinx86) [reply tests/integration_tests/security/test_vulnerabilities.py:183]: What Pablo says is right. In A/B testing, the test would like just to test whether there is difference between A and B. The result of spectre/meltdown checker can fail due to AMI update. Not to be blocked by this type of failure, we introduced A/B testing. So we should ignore the return code here. But let me check whether it ignores the return code if it is not a PR.

Comment (zulinx86) [reply tests/integration_tests/security/test_vulnerabilities.py:183]: I've caught up the discussion here now. Before introduction of A/B testing, this test was run as part of optional PR since it can fail due to AMI updates. My suggestion is to run this test mandatorily on PR but run this separately on nightly test (e.g. run in `.buildkite/pipeline_security.py`) to distinguish from functional test failures.

Comment (bchalios) [reply tests/integration_tests/security/test_vulnerabilities.py:183]: Ok, so IIUC it makes sense to merge this PR to unblock the nightly pipeline and then we can do the suggested change on a follow up?

**Review by bchalios — APPROVED** [review]

**Pull request merged**
